workflow specific caching (#852)
codemagic-ci-cd/codemagic-docs: content/yaml/yaml-getting-started.md

Note: Codemagic doesn't support caching symlinks.
{{</notebox>}}

{{<notebox>}}

> Note: Each workflow has its own caching setup and they are cached separately. It is possible to view them under the Caching tab in the Codemagic UI. 

Reviewer: ```suggestion
Note: Each workflow has its own cache. It is possible to view the cache for each workflow under the Caching tab in the Codemagic UI. 
```
